### DISCUSSION TITLE
Allow Twig string interpolation in renderObjectTemplate

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#9138] [Ideas] [answered]
Well that's basically it.
Currently Craft parses
```
{% set foo = 5 }}
{{ #{foo} }}
```
 into 
```
{% set foo = 5 }}
{{ #{{ (_variables.foo ?? object.foo)|raw }} }}
```
but that results in a runtime error.
I'm not a regex expert but changing your [regex ](https://github.com/craftcms/cms/blob/3.7/src/web/View.php#L708) from

```
(?<!\{)\{\s*(\w+)([^\{]*?)\}
```

to
```
(?<!\{|#)\{\s*(\w+)([^\{]*?)\}
```
Did it for me. (just adding `#` to the negative lookbehind)
I tested a few different scenarios via regex tester and nothing broke, but I'm not entirely sure of all cases.

My use case for this is: we have a project where clients need to insert formulas and we made that via Twig renderObjectTemplate. Sometimes we want to debug some information in those templates via string interpolation but that's not possible and always concatenate strings via `~` is tedious

**Edit**<br>
in case someone really used `#{foo}` on purpose to render `#Bar` you could escape it via `\` so I would use `\#{foo}` or something like that

**Comment by brandonkelly** [comment, marked as answer]
Good idea! I’ve just added support for string interpolation for the next 3.7 Beta release (ef28fa66f8c1a549a16c763acf12c0a690242861).

Actually, we’re just ignoring anything between print tag delimiters (`{{...}}`) now. since it is always invalid to add a print tag inside another one.

**Reply by brandonkelly** [reply]
3.7.0-beta.6 is out now with that change.

**Reply by Anubarak** [reply]
@brandonkelly I'm at the latest version of Craft 3,7.2 but I'm not able to render my template except when I change the regex.
Did I miss anything?

```php
$template = '{% set string = "test #{foo} 5" %}{{string}}';

$html = $this->getView()->renderObjectTemplate($template, null, [
    'foo' => 'bar'
]);;
```

```
A hash key must be followed by a colon (:). Unexpected token "punctuation" of value "}" ("punctuation" expected with value ":") in "__string_template__87ad1497ae54e66ab8560fc74075d08dd7ebc6e9c2b2d45027a456a0c09f58ec" at line 1.
```

edit:
Ahh I see. only inside `{{` not in `{%` tags. Guess I'll either need a workaround or a friendly Brandon that applies this rule to `{%` as well 😇

**Reply by brandonkelly** [reply]
@Anubarak Yeah sorry I was going with your original example that used print tags. Just released 3.7.3 with a fix for normal Twig tags as well.

**Reply by Anubarak** [reply]
Nono, totally my fault for not explaining it properly.
Thank you - as always I appreciate your work 👍
